test/e2e/e2e_suite_test.go

```diff
@@ -55,87 +57,157 @@ var (
+}
+
+var _ = BeforeSuite(func() {
+	ctx = context.Background()
+	managerImage := os.Getenv("MANAGER_IMAGE")
+	if managerImage == "" {
+		managerImage = "gcr.io/k8s-staging-cluster-api/cluster-api-controller:master"
 	}
-	kindCluster.Setup()
-	loadManagerImage(kindCluster)
+	By("setting up in BeforeSuite")
+	var err error
 
-	// Deploy the CAPA components
-	deployCAPIComponents(kindCluster)
+	// Set up the local state of CAPI provider components
+	core := &localCAPIGenerator{}
 
-	kindClient, err = crclient.New(kindCluster.RestConfig(), crclient.Options{Scheme: scheme.SetupScheme()})
-	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	// set up cert manager generator
+	cm := &generators.CertManager{ReleaseVersion: "v0.11.0"}
 
-	// Verify capi components are deployed
-	waitDeployment(capiNamespace, capiDeploymentName)
-}, setupTimeout)
+	scheme := runtime.NewScheme()
+	Expect(corev1.AddToScheme(scheme)).To(Succeed())
+	Expect(clientgoscheme.AddToScheme(scheme)).To(Succeed())
+	Expect(capiv1.AddToScheme(scheme)).To(Succeed())
 
-var _ = ginkgo.AfterSuite(func() {
-	fmt.Fprintf(ginkgo.GinkgoWriter, "Tearing down kind cluster\n")
-	kindCluster.Teardown()
-	os.RemoveAll(suiteTmpDir)
-})
+	// crete kind cluster
+	kindClusterName := "capi-test-" + util.RandomString(6)
+	kindCluster, err = kind.NewCluster(ctx, kindClusterName, scheme, managerImage)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(kindCluster).NotTo(BeNil())
 
-func waitDeployment(namespace, name string) {
-	fmt.Fprintf(ginkgo.GinkgoWriter, "Ensuring %s/%s is deployed\n", namespace, name)
-	gomega.Eventually(
-		func() (int32, error) {
-			deployment := &appsv1.Deployment{}
-			if err := kindClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: namespace, Name: name}, deployment); err != nil {
-				return 0, err
-			}
-			return deployment.Status.ReadyReplicas, nil
-		}, 5*time.Minute, 15*time.Second,
-	).ShouldNot(gomega.BeZero())
-}
+	kindClient, err = kindCluster.GetClient()
+	Expect(err).NotTo(HaveOccurred())
 
-func loadManagerImage(kindCluster kind.Cluster) {
-	if managerImage != nil && *managerImage != "" {
-		kindCluster.LoadImage(*managerImage)
+	// TODO: REMOVE THIS
+	fmt.Println("ALL KNOWN TYPES")
+	for k, v := range scheme.AllKnownTypes() {
+		if k.Group == "cluster.x-k8s.io" {
+			fmt.Printf("%s: %s\n", k, v)
+		}
 	}
-}
 
-func applyManifests(kindCluster kind.Cluster, manifests *string) {
-	gomega.Expect(manifests).ToNot(gomega.BeNil())
-	fmt.Fprintf(ginkgo.GinkgoWriter, "Applying manifests for %s\n", *manifests)
-	gomega.Expect(*manifests).ToNot(gomega.BeEmpty())
-	kindCluster.ApplyYAML(*manifests)
-}
+	// install components
+	framework.InstallComponents(ctx, kindCluster, cm)
+	// TODO: Can the generator handle the install and waiting for its
+	// components. Instead of calling it component generator, we can call it
+	// component installer.
+	framework.WaitForAPIServiceAvailable(ctx, kindCluster, "v1beta1.webhook.cert-manager.io")
```

Comment:
e2e test framework feedback:
As part of component installation we should have a "post-check" to verify components have been installed correctly. Instead of having something like "component generator" we could just create "component installers" that would encapsulate this behavior. That way we can pass the "installers" to `framework.InstallComponents(ctx, kindCluster, certManInstaller, capiInstaller, capdInstaller)`
Each installer would encapsulate post-check behavior like `WaitForAPIServiceAvailable` for `certManInstaller` and `WaitForPodsReadyInNamespace` for `capiInstaller`.

Discuss with @chuckha